WikiWatershed/model-my-watershed
# Unit Conversion: User Preference

## Overview

Adds a user preference for favored unit scheme, which defaults to Metric, but can be set to US Customary. This preference is saved to the user's account. On load and on update, this changes `clientSettings.unit_scheme`, which will be read at run time to decide which units to show.

Connects #3014 

### Demo

![image](https://user-images.githubusercontent.com/1430060/48441017-7328eb80-e758-11e8-80b4-a5bf5212a4ff.png)

## Testing Instructions

* Check out this branch and `bundle.sh`
* Run `manage.sh migrate` to run migrations
* Run `debugserver.sh`
* Go to [:8000/](http://localhost:8000/) and in the browser console, check for `clientSettings.unit_scheme`. Ensure it is 'METRIC'
* Create a new account in MMW. In the registration modal, ensure you see a dropdown for Unit Scheme, which should default to Metric.
* In the browser console, check that `clientSettings.unit_scheme` is still 'METRIC'
* Choose 'US Customary' as the scheme and save the registration. Ensure that `clientSettings.unit_scheme` is now set to `USCUSTOMARY`.
* Go to your account and change your preference there. Ensure it is reflected correctly in the browser console after saving.
* Refresh the page. Ensure your preference is saved correctly and that `clientSettings.unit_scheme` has the correct value.
